WhisperAPI/WhisperAPI/Controllers/ContextController.cs

```diff
@@ -19,13 +20,17 @@ public ContextController(IContexts contexts)
 
         public override void OnActionExecuting(ActionExecutingContext actionExecutingContext)
         {
+            var searchQuery = (SearchQuery)actionExecutingContext.ActionArguments["searchQuery"];
+
+            log4net.ThreadContext.Properties["requestId"] = Guid.NewGuid();
             if (!this.ModelState.IsValid)
             {
                 actionExecutingContext.Result = this.BadRequest(this.ModelState);
+                Log.Error($"ChatKey: {searchQuery.ChatKey}, Query: {searchQuery.Query}, Type: {searchQuery.Type}");
                 return;
             }
 
-            var searchQuery = (SearchQuery)actionExecutingContext.ActionArguments["searchQuery"];
+            Log.Debug($"ChatKey: {searchQuery.ChatKey}, Query: {searchQuery.Query}, Type: {searchQuery.Type}");
```

Comment:
This is a general comment, but it's better in my opinion to log a JSON serialized object if you want to log all of its content. This is because:
1) When those are contracts, we can just copy/paste JSON from the logs in Postman to reproduce the failing case.
2) It's more maintainable, if the object changes, we don't have to update the logging.
3) It's much faster to add the logging statement, something like Log.Debug($"Search query:\r\n{JSONConvert.SerializeObject(query)}");

Finally, make sure that the JSON in the logs is pretty printed, a JSON like:
{
    "Field": "Value",
    "AnotherField": "Value"
}

and not:
{"Field":"Value","AnotherField":"Value"}

If the logging statement becomes too complicated to pretty print, then we can do extension methods on the ILog (public static void DebugObject(this ILog logger)...), but we don't want to do this unless we have a good reason to do it.